spec/controllers/widgets_controller_spec.rb

```diff
@@ -0,0 +1,181 @@
+# coding: utf-8
+require File.expand_path(File.dirname(__FILE__) + '/../spec_helper')
+
+describe WidgetsController do
+
+    include LinkToHelper
+
+    describe "#show" do
+
+        before do
+            @info_request = FactoryGirl.create(:info_request)
+            AlaveteliConfiguration.stub!(:widget_page).and_return(true)
+        end
+
+        it 'should render the widget template' do
+            get :show, :request_id => @info_request.id
+            expect(response).to render_template('show')
+        end
+
+        it 'should find the info request' do
+            get :show, :request_id => @info_request.id
+            assigns[:info_request].should == @info_request
+        end
+
+        it 'should create a track thing for the request' do
+            get :show, :request_id => @info_request.id
+            assigns[:track_thing].info_request.should == @info_request
+        end
+
+        it 'should assign the request status' do
+            get :show, :request_id => @info_request.id
+            assigns[:status].should == @info_request.calculate_status
+        end
+
+        it 'should not send an x-frame-options header' do
+            get :show, :request_id => @info_request.id
+            response.headers["X-Frame-Options"].should be_nil
+        end
+
+        context 'for a non-logged-in user' do
+
+            context 'if no widget-vote cookie is set' do
+
+                it 'should set a widget-vote cookie' do
+                    cookies[:widget_vote].should be_nil
+                    get :show, :request_id => @info_request.id
+                    cookies[:widget_vote].should_not be_nil
+                end
+
+            end
+
+        end
+
+        context 'when widgets are not enabled' do
+
+            it 'should return a 404' do
+                AlaveteliConfiguration.stub!(:widget_page).and_return(false)
+                lambda{ get :show, :request_id => @info_request.id }.should
+                    raise_error(ActiveRecord::RecordNotFound)
+            end
+
+        end
+
+        context "when the request's prominence is not 'normal'" do
+
+            it 'should return a 403' do
+                @info_request.prominence = 'hidden'
+                @info_request.save!
```

Comment:
Could shorten `L67-68` to `@info_request.update_attributes!(:prominence => 'hidden')`